server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/AbstractRedirect.java

```diff
@@ -1025,6 +1025,7 @@ protected void updateHeaders() throws MessagingException {
             setIsReply(newMail, isReply(originalMail), originalMail);
 
             newMail.getMessage().saveChanges();
+            newMail.removeAllAttributes();
```

Comment:
- The bug I described affects other mailets.
- The only way to solve it is to discard mail attributes.

It can be done here, or within MailImpl :: duplicate (that we would call instead of new MailImpl). Or an other method like this. But it do not change what we are doing. It is just about *syntax*.